### PR TITLE
Upload_it notebook extension: open the URL returned by the upload server

### PR DESCRIPTION
**Pull request by salikh**
Also enable authentication of xhr requests.

### CODE REVIEW

File: nbextensions/upload_it/main.js [hunk 1]
```diff
@@ -79,14 +79,26 @@ define([
...
               url: url,
+              xhrFields: {withCredentials: true},
               data: formdata,
               contentType: false,
               processData: false,
               method: "POST",
               success: function(data, status, jqXHR) {
-                window.console.log("Upload OK", data);
+                // Open the report in a new tab.
+                var u = new URL(url);
```
Comment (kw-udon) [inline nbextensions/upload_it/main.js]: style: Let's use `const`.
https://google.github.io/styleguide/jsguide.html#features-use-const-and-let

File: nbextensions/upload_it/main.js [hunk 2]
```diff
@@ -79,14 +79,26 @@ define([
...
+                var reportURL = u.protocol + "//" + u.host + data;
```
Comment (kw-udon) [inline nbextensions/upload_it/main.js]: ditto

File: nbextensions/upload_it/main.js [hunk 3]
```diff
@@ -79,14 +79,26 @@ define([
...
+                window.open(reportURL, '_blank');
+                window.console.log("Upload OK", reportURL);
               },
               error: function(jqXHR, status, err) {
+                if (err == "Unauthorized") {
+                  window.console.log("Unauthorized, attempting login");
+                  let loginURL = new URL(configuration.upload_it_server_url);
```
Comment (kw-udon) [inline nbextensions/upload_it/main.js]: You can use `const` for loginURL as well.

File: nbextensions/upload_it/main.js (hunk 1 above)
Comment (salikh) [reply nbextensions/upload_it/main.js]: Thanks for the pointer. Done.

File: nbextensions/upload_it/main.js (hunk 2 above)
Comment (salikh) [reply nbextensions/upload_it/main.js]: Thank you for the hint. Used  'u.pathname = data'.

File: nbextensions/upload_it/main.js (hunk 3 above)
Comment (salikh) [reply nbextensions/upload_it/main.js]: No, it is modified in the line 'loginURL.pathname = '/login';'

**Comment by salikh** [comment]
Thank you for the review! Pulling now without waiting for another round of review, but feel free to add more comments if you have any, and I will follow up later.

File: nbextensions/upload_it/main.js (hunk 3 above)
Comment (kw-udon) [reply nbextensions/upload_it/main.js]: Not really. `const` just prevents reassignment of variables. [MDN doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const#Description) says 
> the object's contents (e.g., its properties) can be altered.

```javascript
const x = {};
x.foo = "foo"; // OK
x = {}         // Error
```

**Pull request merged**